slurm/pi_aggregation.py

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python3
+from __future__ import print_function, division
+import json
+import sys
+
+def calculate_average_pi(filenames):
+    total_successes = 0
+    total_iterations = 0
+    for filename in filenames:
+        with open(filename, 'r') as f:
+            estimation = json.load(f)
```

Comment:
Do you want to have it search for a particular line that's labeled as the results?  Maybe a prefix of `RESULT: ` or some such?  So that if there are more print statements, or there are more warning/error messages, it still works?

... it's not exactly needed here, but I'm thinking how I would design something to be more robust.  They could all be written to separate files, but if the results can be parsed from the normal stdout that seems like less parts so easier to maintain and use.

Reply:
But isn't it already structured? All the debug messages and errors are now redirected in stderr. For future extensions, maybe JSON is more flexible?
I feel like adding the parsing would maybe add more problems in the future.

Reply:
We talked and yeah, it makes more sense to define the stdout to be a json file, with no extraneous characters.  The other end is a bunch of array tasks which all append to the same file, and stuff is disentangled later (this is a good example becaues it makes fewer files).  But that can be done for some future example.